src/go/rpk/pkg/cli/cmd/cluster/maintenance/enable.go

```diff
@@ -67,16 +75,34 @@ node exists that is already in maintenance mode then an error will be returned.
 			}
 
 			out.MaybeDie(err, "error enabling maintenance mode for node %d: %v", nodeID, err)
-			fmt.Printf("Successfully enabled maintenance mode for node %d\n", nodeID)
+
+			retries := 3
+			for {
+				b, err := client.Broker(cmd.Context(), nodeID)
+				if err != nil {
+					if retries <= 0 {
+						out.Die("Exiting.. error retrieving broker status after emitting the request.")
+					}
+					retries--
+					time.Sleep(time.Second * 2)
+					continue
+				}
+				retries = 3
+				if b.Maintenance.Draining {
+					break
+				}
+				time.Sleep(time.Second * 2)
+			}
 
 			if !wait {
+				fmt.Printf("Successfully enabled maintenance mode for node %d. Check the partition draining status with 'rpk cluster maintenance status'.\n", nodeID)
 				return
+			} else {
+				fmt.Printf("Successfully enabled maintenance mode for node %d. Waiting for node to drain...\n", nodeID)
```

Comment:
[nit / non-blocking] You can drop the else clause since the above condition will return.